src/pyodide-py/pyodide/simple_web_loop.py

```diff
@@ -0,0 +1,203 @@
+        """ Since we cannot block, we just ensure that the future is scheduled. """
+        return asyncio.ensure_future(future)
+
+    #
+    # Scheduling methods: use browser.setTimeout to schedule tasks on the browser event loop.
+    #
+
+    def call_soon(self, callback: Callable, *args, context: contextvars.Context = None):
+        """
+        Schedule the callback callback to be called with args arguments at the next iteration of the event loop.
+        """
+        delay = 0
+        return self.call_later(delay, callback, *args, context=context)
+
+    def call_soon_threadsafe(
+        callback: Callable, *args, context: contextvars.Context = None
+    ):
+        """
+        A thread-safe variant of call_soon().
+
+        Note this function is different from the standard asyncio loop implementation, it is current exactly the same as call_soon
+        """
+        return self.call_soon(callback, *args, context=context)
+
+    def call_later(
+        self,
+        delay: float,
+        callback: Callable,
+        *args,
+        context: contextvars.Context = None
+    ):
+        """
+        Schedule callback to be called after the given delay number of seconds (can be either an int or a float).
+        """
+        from js import setTimeout
+
+        if delay < 0:
+            raise Exception("Can't schedule in the past")
+        h = asyncio.Handle(callback, args, self, context=context)
+        setTimeout(h._run, delay * 1000)
+        return h
+
+    def call_at(
+        self,
+        when: float,
+        callback: Callable,
+        *args,
+        context: contextvars.Context = None
+    ):
+        """
+        Schedule callback to be called at the given absolute timestamp when (an int or a float), using the same time reference as loop.time().
+        """
+        cur_time = self.time()
+        delay = when - cur_time
+        return self.call_later(delay, callback, *args, context=context)
+
+    #
+    # The remaining methods are copied directly from BaseEventLoop
+    #
+
+    def time(self):
+        """Return the time according to the event loop's clock.
+
+        This is a float expressed in seconds since an epoch, but the
+        epoch, precision, accuracy and drift are unspecified and may
+        differ per event loop.
+
+        Copied from BaseEventLoop.time
+        """
+        return time.monotonic()
+
+    def create_future(self):
+        """Create a Future object attached to the loop.
+
+        Copied from BaseEventLoop.create_future
+        """
+        return futures.Future(loop=self)
+
+    def create_task(self, coro, *, name=None):
+        """Schedule a coroutine object.
+
+        Return a task object.
+
+        Copied from BaseEventLoop.create_task
+        """
+        self._check_closed()
+        if self._task_factory is None:
+            task = tasks.Task(coro, loop=self, name=name)
+            if task._source_traceback:
+                del task._source_traceback[-1]
```

Comment:
Could you add a inline commment about this?

Reply:
I don't actually understand why this is here either, which is sort of covered by the comment:

> Copied from BaseEventLoop.create_task

I'll go look at CPython git blame and see if I can figure out what it means.

Reply:
This code was added in this commit:

https://github.com/python/cpython/commit/80f53aa9a0b2af28c9d8052c46b452cccb8e0b41

Specifically:

> The traceback is truncated to hide internal calls in asyncio, show only the traceback from user code.

The conditional `if task._source_traceback:` will be true if and only if the event loop returns `True` from `get_debug()`, which we currently never do.